api/lang/runner.go

```diff
@@ -98,6 +101,48 @@ func getStepPositionResponse(uri string) (*gm.StepPositionsResponse, error) {
 	return stepPositionsResponse, nil
 }
 
+func getImplementationFileList() (*gm.ImplementationFileListResponse, error) {
+	implementationFileListRequest := &gm.Message{MessageType: gm.Message_ImplementationFileListRequest, ImplementationFileListRequest: &gm.ImplementationFileListRequest{}}
+	response, err := GetResponseFromRunner(implementationFileListRequest)
+	if err != nil {
+		logger.APILog.Infof("Error while connecting to runner : %s", err.Error())
+		return nil, err
+	}
+	implementationFileListResponse := response.GetImplementationFileListResponse()
+	return implementationFileListResponse, nil
+}
+
+func putStubImplementation(result *lsp.WorkspaceEdit, filePath string, stepTexts []*gm.ProtoStepValue) error {
+	stubImplementationCodeRequest := &gm.Message{MessageType: gm.Message_StubImplementationCodeRequest, StubImplementationCodeRequest: &gm.StubImplementationCodeRequest{ImplementationFilePath: filePath, Steps: stepTexts}}
+	response, err := GetResponseFromRunner(stubImplementationCodeRequest)
+	if err != nil {
+		logger.APILog.Infof("Error while connecting to runner : %s", err.Error())
+		return err
+	}
+	fileEditResponse := response.GetFileChanges()
+	uri := util.ConvertPathToURI(fileEditResponse.FileName)
```

Comment:
It is a good idea to keep the extra manipulations done to runner responses out of this file. Please move the logic to create workspaceEdits out of this file. If possible, reuse already existing methods to do this.